packages/core/src/request.ts

```diff
@@ -161,6 +163,36 @@ export class Request {
         this.headers = { ...headers };
         this.userData = { ...userData };
         this.handledAt = handledAt as unknown instanceof Date ? (handledAt as Date).toISOString() : handledAt!;
+
+        Object.defineProperties(this.userData, {
+            __crawlee: {
+                // eslint-disable-next-line no-underscore-dangle
+                value: this.userData.__crawlee,
+                enumerable: false,
+                writable: true,
+            },
+            toJSON: {
```

Comment:
actually, why do we need this? to me this feels like a no-op, you are adding `toJSON` to an object, and keeping the exact same behaviour as if it is not there. i really dont think we need more than just the `get/set` pair on `Request` object (plus the `__crawlee` property defined above)

Reply:
if the purpose is to hide things in logs, we should use `inspect.custom` instead, that way it will also have effect on `console.log` calls

Reply:
@mnmkng proposed this solution above, since `apify-client` seems to use `JSON.stringify` when serializing data for the API request to the platform (afaiu). I don't think this `toJSON` is a no-op, as non-enumerable properties don't make it into the `JSON.stringify` output?

Reply:
oh right, i thought the purpose is the exact opposite

Reply:
> since apify-client seems to use JSON.stringify when serializing data for the API request to the platform (afaiu)

That would be nice to know for sure 😄 I don't remember exactly. That's why I said that it will most likely work. Have you tested it?

Reply:
Not sure abt the package structure (and local vs platform behaviour), but when running the debugger locally, I got [here](https://github.com/apify/apify-storage-local-js/blob/6083615fa92987f880e19755fb7d9a3818355e4d/src/resource_clients/request_queue.ts#L234), which works as expected :)

Reply:
Yeah, that's local emulation of client. We need to check the real client as well.

Reply:
In `apify-client-js`, the request is passed in [addRequest](https://github.com/apify/apify-client-js/blob/9ff8029f02640ed61f6a93d7615c2578f8775e50/src/resource_clients/request_queue.ts#L97) -> [httpClient.call](https://github.com/apify/apify-client-js/blob/9ff8029f02640ed61f6a93d7615c2578f8775e50/src/http_client.ts#L124) -> [httpClient._createRequestHandler](https://github.com/apify/apify-client-js/blob/9ff8029f02640ed61f6a93d7615c2578f8775e50/src/http_client.ts#L138) to `axios.request()` (in the `data` field), which seems to respect the `toJSON` overrides. This can be further ensured by serializing the data before passing them to `axios`, but might not be necessary.